src/app/src/components/GeographySelector.jsx

```diff
@@ -76,50 +89,60 @@ const GeographySelector = () => {
                 >
                     Next
                 </Button>
-            </HStack>
-            <Flex alignItems='baseline'>
-                <Button
-                    variant='link'
-                    size='lg'
-                    onClick={() => handleGeoSet(GEO_REGION)}
-                >
-                    REGIONS
-                </Button>
-                <Text
-                    fontSize='sm'
-                    textTransform='uppercase'
-                    mx='2'
-                    color='gray.600'
-                    fontWeight='medium'
-                >
-                    or
-                </Text>
-                <Button
-                    variant='link'
-                    size='lg'
-                    onClick={() => handleGeoSet(GEO_COUNTRY)}
-                >
-                    COUNTRIES
-                </Button>
             </Flex>
-            <VStack>
-                <CheckboxGroup
-                    key={`geogroup-${geoMode}`}
-                    onChange={handleSelection}
-                    defaultValue={currentGeo}
-                >
-                    {geoList.map(geo => (
-                        <Checkbox
-                            key={`geo-${geo}`}
-                            value={geo}
+            <Flex>
+                <Flex flex={1} direction='column'>
+                    <Flex alignItems='baseline' m={4}>
+                        <Button
+                            variant='link'
                             size='lg'
-                            colorScheme='red'
+                            onClick={() => handleGeoSet(GEO_REGION)}
                         >
-                            {geo}
-                        </Checkbox>
-                    ))}
-                </CheckboxGroup>
-            </VStack>
+                            REGIONS
+                        </Button>
+                        <Text
+                            fontSize='sm'
+                            textTransform='uppercase'
+                            mx='2'
+                            color='gray.600'
+                            fontWeight='medium'
+                        >
+                            or
+                        </Text>
+                        <Button
+                            variant='link'
+                            size='lg'
+                            onClick={() => handleGeoSet(GEO_COUNTRY)}
+                        >
+                            COUNTRIES
+                        </Button>
+                    </Flex>
+                    <SearchInput query={query} setQuery={setQuery} />
```

Comment:
Based on the visual designs, I think this only needs to be applied when `COUNTRIES` are selected as the `geoMode`. This will make the filtering logic a little more messy, but I agree that we probably don't need the functionality for regions, which is a small list.

Reply:
Good call. I updated to hide the search when `REGIONS` is selected and to only apply the filtering for `COUNTRIES`. Once the search was hidden for regions, I realized that the list items were misaligned compared to the visual design so I tweaked that a bit.

Reply:
I've set it so that the filter is only applied and the search input only shows when `COUNTRIES` is selected. The alignment of the countries was a bit off so I made a tweak there at the same time.